src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs

```diff
@@ -376,6 +376,37 @@ public static void Main()
 
         }
 
+        [Theory]
+        [InlineData("netcoreapp2.0", true)]
+        [InlineData("netcoreapp3.0", true)]
+        [InlineData("net5.0", true)]
+        [InlineData("net6.0", false)]
+        public void It_stops_generating_runtimeconfig_dev_json_after_net6(string targetFramework, bool generateRuntimeConfigDevJson)
```

Comment:
Can we also check that the file is indeed generated if `("net6.0", true)` and that using the new property `GenerateRuntimeConfigDevFile` has no effect on previews versions?

Reply:
This helped me realize I needed to only set the property if it wasn't already set, thanks for that. Added more tests as well.

Reply:
Just realized I implemented it so the user-set property has the final say. You're saying no matter what, if the user sets `GenerateRuntimeConfigDevFile` to true and targets < net6.0, we should still generate the file?

Reply:
Sorry for the delay. `GenerateRuntimeConfigDevFile` should be false by default for .NET 6 onwards, and true otherwise. All previous versions should be generating this file by default. I think it's fine if we allow previous versions to stop generating the file whenever a user sets `<GenerateRuntimeConfigDevFile>false</GenerateRuntimeConfigDevFile>`.

Reply:
Perfect. The change is good to merge then 👍